src/templates/template-vision-classification/main.py

```diff
@@ -50,7 +49,7 @@ def run(local_rank: int, config: Any):
     # print training configurations
     logger = setup_logging(config)
     logger.info("Configuration: \n%s", pformat(vars(config)))
-    (config.output_dir / "config-lock.yaml").write_text(yaml.dump(config))
+    write_config_to_output_dir(config)
```

Comment:
This comment is related to this PR and also to the previous code. How do we ensure that only one rank is writing to the disk avoiding data corruption if all ranks start to write into the same file together ?

Reply:
Yes, but right now it is not yet the case, right ?

Reply:
Yes, made the changes!